lib/runtime/PublicPathRuntimeModule.js

```diff
@@ -18,14 +18,22 @@ class PublicPathRuntimeModule extends RuntimeModule {
 	 * @returns {string} runtime code
 	 */
 	generate() {
-		return `${RuntimeGlobals.publicPath} = ${JSON.stringify(
-			this.compilation.getPath(
-				this.compilation.outputOptions.publicPath || "",
-				{
-					hash: this.compilation.hash || "XXXX"
+		const publicPath = this.compilation.outputOptions.publicPath || "auto";
+		if (publicPath === "auto") {
+			return `${RuntimeGlobals.publicPath} = (() => {
+				if ("currentScript" in window.document) {
```

Comment:
I see tests are failing because of this line. What's the right way to handle this? Should I change the condition to the following?

```js
window.document && "currentScript" in window.document
```